bleu/cowswap
# Jean/cow 346 airdrop hook UI refactoring based on designs

# Summary

- Add Claim Airdrop Hook as a PreHook app.
- Add 2 example airdrops in order to test the application: [COW](https://github.com/bleu/cow-airdrop-contract-deployer/tree/example) and 'YYY', which is just a bad example that returns an error

![image](https://github.com/user-attachments/assets/806e50a0-0539-49d8-a832-955c07dc21d2)

![image](https://github.com/user-attachments/assets/429bd4e6-9c25-46fc-8ba8-a12c22e73779)

![image](https://github.com/user-attachments/assets/416f33dd-e43f-4df6-85f1-4867753919fd)

![image](https://github.com/user-attachments/assets/f03a4d7e-a362-4b2a-99d1-50e15be99063)

![image](https://github.com/user-attachments/assets/ce6c3e30-720d-45ef-bba1-79cbb980b6f6)

![image](https://github.com/user-attachments/assets/6f242e8e-3273-480e-913c-7fa34c83fa4e)



# To Test

1. Open the page `/#/11155111/swap/hooks/WETH`

- You should be able to see the swap with hooks screen

2. Connect your wallet and switch to Sepolia Network

3. Click 'Add PreHook action' > 'Claim Airdrop' and test the app

4. If your wallet is in [this csv](https://github.com/bleu/cow-airdrop-contract-deployer/blob/example/input-folder/allocations.csv) you should be able to see your claimable tokens. Else, you will probably get the message 'No tokens to claim'
